pkg/gds/gds_test.go

```diff
@@ -211,44 +211,73 @@ func (s *gdsTestSuite) TestLookup() {
 	require := s.Require()
 	ctx := context.Background()
 
-	charlieVASP, err := s.fixtures.GetVASP("charliebank")
-	require.NoError(err)
-
 	// Start the gRPC client
 	require.NoError(s.grpc.Connect(ctx))
 	defer s.grpc.Close()
 	client := api.NewTRISADirectoryClient(s.grpc.Conn)
 
-	// Supplied VASP ID does not exist
-	request := &api.LookupRequest{
-		Id: "abc12345-41aa-11ec-9d29-acde48001122",
-	}
-	_, err = client.Lookup(ctx, request)
-	require.Error(err)
-
-	expected := &api.LookupReply{
-		Id:                  charlieVASP.Id,
-		RegisteredDirectory: charlieVASP.RegisteredDirectory,
-		CommonName:          charlieVASP.CommonName,
-		Endpoint:            charlieVASP.TrisaEndpoint,
-		IdentityCertificate: charlieVASP.IdentityCertificate,
-		Country:             charlieVASP.Entity.CountryOfRegistration,
-		VerifiedOn:          charlieVASP.VerifiedOn,
-		Name:                "CharlieBank",
-	}
-
-	// VASP exists in the database
-	request.Id = charlieVASP.Id
-	reply, err := client.Lookup(ctx, request)
-	require.NoError(err)
-	require.True(proto.Equal(expected, reply))
-	// TODO: Check that a signing certificate is returned
+	s.Run("IDNotFound", func() {
+		// Supplied VASP ID does not exist
+		require := s.Require()
+		request := &api.LookupRequest{
+			Id: "abc12345-41aa-11ec-9d29-acde48001122",
+		}
+		_, err := client.Lookup(ctx, request)
+		require.EqualError(err, "rpc error: code = NotFound desc = could not find VASP by ID")
+	})
```

Comment:
I like the isolation of the subtests; it makes it way easier to read!

Reply:
I agree that the other mega endpoint tests could use a test breakdown like this - it makes it far easier to debug the tests too since each test is run independently.